ToC and Script for Video 13: Thing Description (#177)
w3c/wot-cg: Tutorials/whatiswot/13-Thing_Description/README.md

## Table of Contents

- Introduction and Definition
- TD Components
- Interactions
- Protocols

## Preliminary Script

### Introduction and Definition

> - In this video we will provide a general overview of a fundamental concept in the Web of Things - the Thing Description, or TD. A TD acts as a unique blueprint for the respective Thing, offering a standardized way to describe its functionality and how to interact with it.
> 
> - Thing Descriptions are written using the JSON-LD format, making it both machine and human-readable. To understand this better, let's take a look at a simple example - a smart coffee machine, and break down the components in its Thing Description:
> 
> **Code: Complete Thing Description - Coffee Machine Example**
> 

Reviewer: ```suggestion
- In this video, we will provide a general overview of a fundamental concept in the Web of Things — the Thing Description, or TD. A TD acts as a unique blueprint for the respective Thing, offering a standardized way to describe its functionality and how to interact with it.

- Thing Descriptions are written using the JSON-LD format, making it both machine and human-readable. To understand this better, let's take a look at a simple example — a smart coffee machine — and break down the components in its Thing Description:

**Code: Complete Thing Description — Coffee Machine Example**

```

```json
{
    "@context": ["https://www.w3.org/2019/wot/td/v1"], 

>     "id": "urn:dev:ops:Coffee-Machine-1234",

Reviewer 2: ```suggestion
    "id": "urn:uuid:0804d572-cce8-422a-bb7c-4412fcd56f06",
```

    "title": "Smart Coffee Machine",
    "description" : "Simplified example of a smart coffee machine",
    "securityDefinitions": { ... },
    "security": { ... },
    "properties": {
        "coffeeBeansLeft": { ... }
    },
    "actions": {
        "brewCoffee": { ... }
    },
    "events": {
        "lowOnWater": { ... }
    },
    "links": [ ... ]
}
```

### TD Components

> **Code: Complete Thing Description - Coffee Machine Example**
> 
> <!-- Mark each component on the graphic example and explain what they do briefly -->

Reviewer: ```suggestion
**Code: Complete Thing Description — Coffee Machine Example**

<!-- Mark each component on the graphic example and briefly explain what they do -->
```

> - The Thing Descriptions begins with the Thing's metadata. It provides the essential information about the device, like its unique ID, title and description.

Reviewer: ```suggestion
- The Thing Descriptions begins with the Thing's metadata. It provides the essential information about the device, like its unique ID, title, and description.
```

> - Then in the Definitions of Interaction Affordances we specify ways you can interact with the device, through its various properties, actions and events. The example here shows the property coffeeBeansLeft, which describes the current state of the coffee beans, the action brewCoffee, which triggers brewing and the event lowOnWater, which alerts you when the machine is low on water. We’ll explore Interaction Affordances in greater detail in the next video of this series. 

Reviewer: ```suggestion
- Then, in the Definitions of Interaction Affordances, we specify ways you can interact with the device, through its various properties, actions, and events. The example here shows the property `coffeeBeansLeft`, which describes the current state of the coffee beans; the action `brewCoffee`, which triggers brewing; and the event `lowOnWater`, which alerts you when the machine is low on water. We’ll explore Interaction Affordances in greater detail in the next video of this series. 
```

> - Then, in the Definitions of Interaction Affordances, we specify ways you can interact with the device, through its various properties, actions, and events. The example here shows the property `coffeeBeansLeft`, which describes the current state of the coffee beans; the action `brewCoffee`, which triggers brewing; and the event `lowOnWater`, which alerts you when the machine is low on water. We’ll explore Interaction Affordances in greater detail in the next video of this series. 

Reviewer 2: ```suggestion
- Then, in the definitions of Interaction Affordances, we specify ways a Consumer application can interact with the Thing, through its various properties, actions, and events exposed on its network interfaces. The example here shows the property `coffeeBeansLeft`, which describes the current state of the coffee beans; the action `brewCoffee`, which triggers brewing; and the event `lowOnWater`, which alerts the Consumer when the machine is low on water. We’ll explore Interaction Affordances in greater detail in the next video of this series. 
```

> - The Security Metadata defines security requirements and mechanisms, that ensure secure access to the device by the user.

Reviewer 2: ```suggestion
- The Security Metadata defines security mechanisms required to ensure secure access to the Thing by the Consumer application. This way the Consumer application knows what type of credentials are needed to execute different operations.
```

> - To enable universal understanding, we can add Semantic Annotations. These allow other systems to interpret the device’s terms and functions consistently.

Reviewer: ```suggestion
- To enable universal understanding, we can add Semantic Annotations. These allow other systems to consistently interpret the device’s terms and functions.
```

> - To enable universal understanding, we can add Semantic Annotations. These allow other systems to consistently interpret the device’s terms and functions.

Reviewer 2: ```suggestion
- To add keywords that are not part of the TD standard, we can add Semantic Annotations. These allow other systems to consistently interpret the device’s terms and functions. Here, we annotate using `schema.org` to add the name of the manufacturer, which is not part of the core TD specification.
```

> - Finally, we can also link the Thing Description to other documents, such as documentation or a user interface.

Reviewer: ```suggestion
- Finally, we can link the Thing Description to other documents, such as documentation or a user interface.
```
